hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/DistributedScheduledExecutorService.java

```diff
@@ -0,0 +1,157 @@
+import com.hazelcast.spi.ManagedService;
+import com.hazelcast.spi.MigrationAwareService;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.PartitionMigrationEvent;
+import com.hazelcast.spi.PartitionReplicationEvent;
+import com.hazelcast.spi.RemoteService;
+import com.hazelcast.spi.impl.executionservice.InternalExecutionService;
+import com.hazelcast.spi.partition.MigrationEndpoint;
+
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class DistributedScheduledExecutorService
+        implements ManagedService, RemoteService, MigrationAwareService {
+
+    public static final String SERVICE_NAME = "hz:impl:scheduledExecutorService";
+
+    public static final int MEMBER_PARTITION = -1;
+
+    private NodeEngine nodeEngine;
+
+    private ScheduledExecutorPartition[] partitions;
+
+    private ScheduledExecutorPartition memberPartition;
+
+    private final ConcurrentMap<String, Boolean> shutdownExecutors
+            = new ConcurrentHashMap<String, Boolean>();
+
+    public DistributedScheduledExecutorService() {
+    }
+
+    @Override
+    public void init(NodeEngine nodeEngine, Properties properties) {
+        int partitionCount = nodeEngine.getPartitionService().getPartitionCount();
+        this.nodeEngine = nodeEngine;
+        this.partitions = new ScheduledExecutorPartition[partitionCount];
+        reset();
+    }
+
+    public ScheduledExecutorPartition getPartition(int partitionId) {
+        if (partitionId == MEMBER_PARTITION) {
+            return memberPartition;
+        }
+
+        return partitions[partitionId];
+    }
+
+    public NodeEngine getNodeEngine() {
+        return nodeEngine;
+    }
+
+    @Override
+    public void reset() {
+        shutdown(true);
+
+        memberPartition = new ScheduledExecutorPartition(nodeEngine, MEMBER_PARTITION);
+
+        for (int partitionId = 0; partitionId < partitions.length; partitionId++) {
+            if (partitions[partitionId] != null) {
+                partitions[partitionId].destroy();
+            }
+            partitions[partitionId] = new ScheduledExecutorPartition(nodeEngine, partitionId);
+        }
+    }
+
+    @Override
+    public void shutdown(boolean terminate) {
+        shutdownExecutors.clear();
+
+        if (memberPartition != null) {
+            memberPartition.destroy();
+        }
+
+        for (int partitionId = 0; partitionId < partitions.length; partitionId++) {
+            if (partitions[partitionId] != null) {
+                partitions[partitionId].destroy();
+            }
+        }
+    }
+
+    @Override
+    public DistributedObject createDistributedObject(String name) {
+        return new ScheduledExecutorServiceProxy(name, nodeEngine, this);
+    }
+
+    @Override
+    public void destroyDistributedObject(String name) {
+        if (shutdownExecutors.remove(name) != null) {
```

Comment:
Well as per idea behind the design yes :), but in practise no, the Durable Executor service can shutdown the executor too, since I am delegating my work to the same executors, therefore as @gurbuzali suggested this needs fix.